app/src/main/java/com/calculator/MainActivity.java

```diff
@@ -1,16 +1,480 @@
+                    result = "";
+                }
+            }
+        });
+
+        btn_eraseAll.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                values = "";
+                result = "";
+                label_values.setText(values);
+                label_result.setText(String.valueOf(result));
+            }
+        });
+
+        btn_point.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                values = values + ".";
+                label_values.setText(values);
+            }
+        });
+
+        btn_parenthesisOpen.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                values = values + "(";
+                label_values.setText(values);
+            }
+        });
+
+        btn_parenthesisClose.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                values = values + ")";
+                label_values.setText(values);
+            }
+        });
+
+        switch_mode.setOnClickListener(new View.OnClickListener() {
+            @Override
+            public void onClick(View view) {
+                if (switch_mode.isChecked()) {
+                    Intent intent = new Intent(MainActivity.this, AdvancedCalcActivity.class);
+                    switch_mode.setText(R.string.modeAdvanced);
+                    startActivityForResult(intent,ACTIVITY2);
+                }
+            }
+        });
+    }
+
+    @Override
+    protected void onActivityResult(int requestCode, int resultCode, @Nullable Intent data) {
+        super.onActivityResult(requestCode, resultCode, data);
+
+        if(requestCode == ACTIVITY2) {
+            if(resultCode == RESULT_OK) {
+                switch_mode.setChecked(false);
+                switch_mode.setText(R.string.modeBasic);
+            }
+        }
+    }
+
+    @Override
+    protected void onResume() {
+        super.onResume();
+        // The activity has become visible (it is now "resumed").
+    }
+
+    @Override
+    protected void onPause() {
+        super.onPause();
+        // Another activity is taking focus (this activity is about to be "paused").
+    }
+
+    @Override
+    protected void onStop() {
+        super.onStop();
+        // The activity is no longer visible (it is now "stopped")
+    }
+
+    @Override
+    protected void onDestroy() {
+        super.onDestroy();
+        // The activity is about to be destroyed.
+    }
+
+    private static String calc(String expression) {
+        expression = prepareExpression(expression);
+        int pos = 0;
+        //Extracting expression from braces, doing recursive call
+        //replace braced expression on result of it solving
+        if (-1 != (pos = expression.indexOf("("))) {
```

Comment:
If you have a long expression in a if block statement you could add these operations into a method for example:

```
if(isValidExpression("("))
....
   private boolean isValidExpression(String expression) {
      return -1 != (expression.indexOf(expression)
   }
and for
position = getPosition(expression);

private int getPosition(String expression) {
  return expression.indexOf(expression);
}
```